server/container_create.go

```diff
@@ -1079,8 +1079,17 @@ func (s *Server) createSandboxContainer(ctx context.Context, containerID string,
 		specgen.AddMount(mnt)
 	}
 
-	// Bind mount /etc/hosts for host networking containers
-	if hostNetwork(containerConfig) {
+	isInCRIMounts := func(dst string, mounts []*pb.Mount) bool {
+		for _, m := range mounts {
+			if m.ContainerPath == dst {
+				return true
+			}
+		}
+		return false
+	}
+
+	if !isInCRIMounts("/etc/hosts", containerConfig.GetMounts()) && hostNetwork(containerConfig) {
```

Comment:
Wouldn't we want similar handling if k8s suggested a `resolv.conf` mount, for example?  It might be more robust to setup whatever our default mounts are and loop over `containerConfig.GetMounts()`, clobbering anything that shares their destination.

Also, who is actually calling `mount(2)` for the k8s mounts?  Do we not need `mounts` config entries for them?

Reply:
The mounts are performed by the runtime (runc for e.g.). We just add them to the configuration here. This change looks fairly low risk as we just added one additional guard on top of what we had before.